test/tstDistributedTree.cpp

```diff
@@ -28,8 +28,22 @@
 namespace tt = boost::test_tools;
 
 using PairIndexRank = Kokkos::pair<int, int>;
+using PairRankIndex = Kokkos::pair<int, int>;
```

Comment:
Absolutely not.  Aliasing `Kokkos::pair` was (probably me) already being lazy, this does not protect us against bugs and is awfully confusing.
Any good reason to have both `PairIndexRank` and `PairRankIndex`?

Reply:
As I said before, I need to work more on the tests. Here, I wanted the reverse order to make sure that the new test really executes the callback version properly. I would have gotten the same result ignoring the callback otherwise.